### PR TITLE
Fix auto download permissions

### LINKED ISSUE #69: Automate downloading on mass-cli1.
The recent Mass PR works (#63), yay, but there are some issues, notably with permissions, and downloading via cron.

### Permissions:
We need to make downloaded data accessible from anyone. At the moment, whoever first downloads a jobID, creates and owns the directory in `/gws/nopw/j04/ukesm/BGC_data/`. However, this means that if someone else wants to update that jobID download with recent data, they can not. The solution is to put `chmod g+w; command in the python script.

### Downloading via crontab
So, for some reason, it looks like the mass-cli1 cronjob is run by root, who can't run `moo get`, so we need to `source /etc/bash`. Following the suggestion of Alan, from the jasmin help desk. (Thanks Alan!)

This is already done and will be submitted in a PR soon.

### PR DESCRIPTION
**Pull request by ledm**
This fixes some issues with the auto-downloading of data overnight.

Closes #69.

### CODE REVIEW

File: bgcval2/download_from_mass.py
Comment (ledm) [inline bgcval2/download_from_mass.py:426]: Delete commented line.
```suggestion
```

**Comment by ledm** [comment]
BTW, this is a nightly run thing, and I can't test whether it has worked until tomorrow morning, so I won't merge until then. but @tillku did ask about this before Christmas.

File: bgcval2/download_from_mass.py
Comment (valeriupredoi) [inline bgcval2/download_from_mass.py:425]: is this a script that lives on `/etc` on mass-cli node?

**Review by valeriupredoi — APPROVED** [review]
not sure what that bashrc script is, man, all else looks good to me tho :beer:

**Comment by ledm** [comment]
Cheers V. The etc bashrc on masscli1 loads the moo get tool. And probably some other stuff.  Unfortunately there's no module load on masscli1, otherwise I'd do that. 

Like i said though, not sure if this will work until tomorrow though, so please hold off the merge until then!

**Comment by valeriupredoi** [comment]
> Like i said though, not sure if this will work until tomorrow though, so please hold off the merge until then!

You usually merge your own PRs so no worries there - we can actually make it a rule - something you implement, I have a look and approve or suggest changes, and you merge - you are in the best position to do that since this repo is 85% your shtuff, man :smiley: 

About the script (`/etc/bashrc`) - it seems overhead-y and quite unsecured too, obv not your fault, but rather, JASMIN admins', maybe we should just make a copy of that locally and execute it from here, rather than execute some very core system script from a user's Python runtime. At any rate, let's see if it works this way, then we can merge andthen think about the more secure option.

File: bgcval2/download_from_mass.py
Comment (ledm) [reply bgcval2/download_from_mass.py:425]: Yeah, its only about 90 lines long, but it's a bit too bash-y for me to understand. I'm happy to follow advice here instead of trying to DIY it! This method ensures that it stays up to date if the system updates /etc/bashrc.

**Comment by ledm** [comment]
Worked great overnight! Merging.

**Comment by ledm** [comment]
Forgot to add this behaviour to the documentation. I'll write something now and merge it.

File: bgcval2/download_from_mass.py
Comment (valeriupredoi) [reply bgcval2/download_from_mass.py:425]: > This method ensures that it stays up to date if the system updates

Yes, good point! There's that, which I thought about too yesterday. Maybe I should have a closer look at the JASMIN setup and have a chat done with Alan/Ag about this

**Comment by tillku** [comment]
So, how is this automatic download from mass supposed to work? I thought I just needed to run my analysis_compare, and then the automated download would pick up the runIDs from that. Is that true? I've tried it and it doesn't work, there's no recent downloads happening as far as I can see... ?

**Comment by ledm** [comment]
Yes, that's the process, but first you need to merge the origin/main branch into your local code:
```
git fetch
git merge origin/main
```

It doesn't look like any of your automated download scrips have changed since the 9th of January? They should be automatically created in /gws/nopw/j04/esmeval/bgcval2/shared_mass_scripts

**Comment by tillku** [comment]
Hmmm ... I'm /sure/ I did the fetch and the merge. I'll double-check. 

No, the automated download scrips have not changed since the 9th of January, I don't need any other runs at the moment

**Comment by ledm** [comment]
Once you have done the merge, you'll also need to run `analysis_compare -y comparison_script.yml` again. This should add the fixes to your download scripts in `/gws/nopw/j04/esmeval/bgcval2/shared_mass_scripts`.

**Comment by tillku** [comment]
Works fine now!

**Pull request merged**
